### PR TITLE
LPD-23255 | LPD-23250 | LPD-23251 | LPD-23252 | LPD-31826

### PR DESCRIPTION
**Pull request by ChrisKian**
Resent from https://github.com/liferay-appsec/liferay-portal/pull/1887 since rebasing also included hundreds of other commits.
Original message:

Pr related to story [LPD-6829](https://liferay.atlassian.net/browse/LPD-6829) Reset scim client provisioning data. [... removed for due to irrelevancy...] See ticket description for ACs

If testing locally, be sure to build portal-language as well, otherwise the locators related to the Reset button and Tooltip will not be found in Playwright.

### CODE REVIEW

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
ci:test:playwright

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
ci:test:playwright

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
Hello @rafaprax, can you please review this SCIM related PR?

Please let me know if you have any questions, thanks!

CC @cdbm

File: modules/dxp/apps/scim/scim-configuration-web/src/main/java/com/liferay/scim/configuration/web/internal/portlet/action/SaveScimConfigurationMVCActionCommand.java [hunk 1]
```diff
@@ -137,6 +150,55 @@ protected void doProcessAction(
...
+			_oAuth2ApplicationLocalService.deleteOAuth2Application(
+				oAuth2Application);
+
+			Indexer<User> userIndexer = IndexerRegistryUtil.nullSafeGetIndexer(
+				User.class);
+
+			Indexer<UserGroup> userGroupIndexer =
+				IndexerRegistryUtil.nullSafeGetIndexer(UserGroup.class);
+
+			ActionableDynamicQuery query =
```
Comment (rafaprax) [inline modules/dxp/apps/scim/scim-configuration-web/src/main/java/com/liferay/scim/configuration/web/internal/portlet/action/SaveScimConfigurationMVCActionCommand.java:170]: @cdbm @ChrisKian rename `query` to `actionableDynamicQuery`, please.

File: modules/dxp/apps/scim/scim-configuration-web/src/main/java/com/liferay/scim/configuration/web/internal/portlet/action/SaveScimConfigurationMVCActionCommand.java [hunk 2]
```diff
@@ -137,6 +150,55 @@ protected void doProcessAction(
...
+				_expandoValueLocalService.getActionableDynamicQuery();
+
+			query.setAddCriteriaMethod(
+				dynamicQuery -> {
+					Property dataProperty = PropertyFactoryUtil.forName("data");
+
+					dynamicQuery.add(dataProperty.eq(scimClientId));
```
Comment (rafaprax) [inline modules/dxp/apps/scim/scim-configuration-web/src/main/java/com/liferay/scim/configuration/web/internal/portlet/action/SaveScimConfigurationMVCActionCommand.java:177]: Hey guys, at first glance this criteria seems weak. Correct me if I'm wrong but are you deleting an `expandoRow` just verifying that the `data` column value has the `scimClientId`? I'm concerned about deleting rows that are not related to this feature. Can we add more restrictions/conditions to the query to ensure we delete the row related to the SCIM client configuration?

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
ci:test:playwright

File: modules/dxp/apps/scim/scim-configuration-web/src/main/java/com/liferay/scim/configuration/web/internal/portlet/action/SaveScimConfigurationMVCActionCommand.java [hunk 3]
```diff
@@ -170,11 +172,31 @@ else if (Objects.equals(cmd, "reset")) {
 			ActionableDynamicQuery actionableDynamicQuery =
 				_expandoValueLocalService.getActionableDynamicQuery();
 
+			ExpandoColumn userScimClientExpandoColumn =
```
Comment (rafaprax) [inline modules/dxp/apps/scim/scim-configuration-web/src/main/java/com/liferay/scim/configuration/web/internal/portlet/action/SaveScimConfigurationMVCActionCommand.java:175]: Hey @cdbm, move the expandoColumn variables closer to their usage and put the condition related to the `columnId` before the `data` condition just to follow the alphabetical order. At the end, the code should be like this

```
actionableDynamicQuery.setAddCriteriaMethod(
				dynamicQuery -> {
					ExpandoColumn userScimClientExpandoColumn =
						_expandoColumnLocalService.getColumn(
							themeDisplay.getCompanyId(), User.class.getName(),
							"CUSTOM_FIELDS", "scimClientId");

					ExpandoColumn userGroupScimClientExpandoColumn =
						_expandoColumnLocalService.getColumn(
							themeDisplay.getCompanyId(),
							UserGroup.class.getName(), "CUSTOM_FIELDS",
							"scimClientId");

					Property columnProperty = PropertyFactoryUtil.forName(
						"columnId");

					dynamicQuery.add(
						columnProperty.in(
							new long[] {
								userScimClientExpandoColumn.getColumnId(),
								userGroupScimClientExpandoColumn.getColumnId()
							}));

					Property dataProperty = PropertyFactoryUtil.forName("data");

					dynamicQuery.add(dataProperty.eq(scimClientId));
				});

```

After this change, please check the test results and if they are good, forward this to BChan

Thanks

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
Hey @rafaprax,
It looks like relevant and stable tests are now truly passing.  Would you mind reviewing and forward after your approval?  Thanks!

CC @cdbm and @daiazius

**Comment by rafaprax** [comment]
ci:forward

**Pull request closed without merging**